eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BitcoinCoreClient.scala

```diff
@@ -42,13 +46,47 @@ import scala.util.{Failure, Success, Try}
 
 /**
  * The Bitcoin Core client provides some high-level utility methods to interact with Bitcoin Core.
+ *
+ * @param rpcClient             bitcoin core JSON rpc client
+ * @param onchainKeyManager_opt optional onchain key manager. If provided and its wallet name matcher our rpcClient wallet's name, it will be used to sign transactions (it is assumed that bitcoin
+ *                              core uses a watch-only wallet with descriptors generated by Eclair with this onchain key manager)
  */
-class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient) extends OnChainWallet with Logging {
+class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient, val onchainKeyManager_opt: Option[OnchainKeyManager] = None) extends OnChainWallet with Logging {
 
   import BitcoinCoreClient._
 
   implicit val formats: Formats = org.json4s.DefaultFormats
 
+  val useEclairSigner = onchainKeyManager_opt.exists(m => rpcClient.wallet.contains(m.wallet))
+
+  //------------------------- WALLET  -------------------------//
+  def importDescriptors(descriptors: Seq[Descriptor])(implicit ec: ExecutionContext): Future[Boolean] = {
+    rpcClient.invoke("importdescriptors", descriptors).collect {
+      case JArray(results) => results.forall(item => {
+        val JBool(success) = item \ "success"
+        val JArray(_) = item \ "warnings"
+        success
+      })
+    }
+  }
+
+  def createDescriptorWallet()(implicit ec: ExecutionContext): Future[Boolean] = {
+    onchainKeyManager_opt match {
+      case None => Future.successful(true) // no eclair-backed wallet is configured
+      case Some(onchainKeyManager) if !rpcClient.wallet.contains(onchainKeyManager.wallet) => Future.successful(true) // configured wallet has a different name
```

Comment:
Should we produce a warning if a wallet exists with a different name and the onchainKeyManager is defined? The user might expect the defined descriptor wallet is used instead of the current wallet.